# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.h
# Commit: fc9ffe1ae7ee
# Full Hash: fc9ffe1ae7ee592931f6adfb3d490ada4dbce43d
# Author: stransky <stransky@redhat.com>
# Date: 2024-07-08 16:33:59
# Regressor Bug: 1898476
# File Overlap Count: 1
# Description:
#   Bug 1898476 [Wayland] Provide surface lock by GtkCompositorWidget r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D214884
# ==============================================================================

diff -r 3e6bd6c75492 -r fc9ffe1ae7ee widget/gtk/nsWindow.h
--- a/widget/gtk/nsWindow.h	Mon Jul 08 08:01:08 2024 +0000
+++ b/widget/gtk/nsWindow.h	Mon Jul 08 08:01:08 2024 +0000
@@ -13,6 +13,7 @@
 
 #include "CompositorWidget.h"
 #include "MozContainer.h"
+#include "MozContainerSurfaceLock.h"
 #include "VsyncSource.h"
 #include "mozilla/EventForwards.h"
 #include "mozilla/Maybe.h"
@@ -421,6 +422,8 @@
 
   static nsWindow* GetFocusedWindow();
 
+  mozilla::UniquePtr<MozContainerSurfaceLock> LockSurface();
+
 #ifdef MOZ_WAYLAND
   // Use xdg-activation protocol to transfer focus from gFocusWindow to aWindow.
   static void TransferFocusToWaylandWindow(nsWindow* aWindow);
